analyzers/tests/SonarAnalyzer.Test/Rules/LoggingArgumentsShouldBePassedCorrectlyTest.cs

```diff
@@ -110,19 +139,17 @@ public void Method(ILogger logger, string message, Exception e)
                     logger.{{methodName}}(CultureInfo.CurrentCulture, message, e);        // Noncompliant
                                                                                           // Secondary @-1
                     logger.{{methodName}}(e, message);                                    // Compliant
-                    logger.{{methodName}}(e, message, e);                                 // Noncompliant
```

Comment:
What if someone passes `null` to the parameter that `Filter` is looking on?

E.g. for this example:

```csharp
logger.{{methodName}}(null, message, e);                                 // FN (?)
```

To track it you would probably need to use `SemanticModel.GetConstantValue` on the argument that matches the `knownType` you are using `Filter` on, but that would mean passing the `invocation` and finding the index of that `ParameterSymbol`. 
Which is not impossible if you use `CSharpFacade.Instance.Method.MethodParameterLookup`.

...but I think it's not that worth it. You should still add a case for it to document the `// FN`